lisdf/components/control.py

```diff
@@ -0,0 +1,108 @@
+#! /usr/bin/env python3
+# -*- coding: utf-8 -*-
+# File   : control.py
+# Author : Jiayuan Mao
+# Email  : maojiayuan@gmail.com
+# Date   : 03/23/2022
+#
+# This file is part of lisdf.
+# Distributed under terms of the MIT license.
+
+from dataclasses import dataclass
+
+import numpy as np
+
+from lisdf.utils.typing import Vector2f, Vector3f
+
+from .base import StringConfigurable
+
+__all__ = [
+    "ControlInfo",
+    "JointInfo",
+    "FixedJointInfo",
+    "HingeJointInfo",
+    "PrismaticJointInfo",
+]
+
+
+@dataclass
+class ControlInfo(object):
+    limited: bool
+    range: Vector2f
+
+    def __init__(self, limited: str, range: Vector2f):
+        assert limited in ("true", "false")
+        self.limited = limited == "true"
+        self.range = range
+
+
+@dataclass
+class JointInfo(StringConfigurable):
+    type_mapping = dict()  # type: ignore
+
+    def __init_subclass__(cls, type: str, **kwargs):
+        super().__init_subclass__(**kwargs)
+        setattr(cls, "type", type)
```

Comment:
Can you explain what's going on here? Looks cool but I haven't seen this pattern used before

Reply:
Added a doc